hawtio-system/src/main/java/io/hawt/web/FileUploadFilter.java

```diff
@@ -0,0 +1,171 @@
+package io.hawt.web;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.lang.StrictMath.toIntExact;
+
+public class FileUploadFilter {
+
+    private static final transient Logger LOG = LoggerFactory.getLogger(FileUploadFilter.class);
+    public static final String FILE_UPLOAD_PROPNAME = "hawtio.config.fileUpload";
+    private List<MagicNumberFileFilter> filters;
+
+    private void addFilters(MagicNumberFileFilter config) {
+        if (filters == null) {
+            this.filters = new ArrayList<>();
+        }
+
+        filters.add(config);
+    }
+
+    public FileUploadFilter() {
+
+        String config = System.getProperty(FILE_UPLOAD_PROPNAME);
+        if (config != null) {
+            LOG.info("Configuring file upload using {} configurations", config);
+
+            try {
+                String[] var0 = config.split(";");
+                for (int i = 0; i <= var0.length - 1; i++) {
+                    MagicNumberFileFilter filter = new MagicNumberFileFilter();
+                    String[] var1 = var0[i].split(",");
+                    for (int j = 0; j <= var1.length - 1; j++) {
+                        if (var1[j].toLowerCase().contains("signature=")) {
+                            filter.setMagicNumbers(hexStringToByteArray(var1[j].replace("signature=", "")));
+                        }
+
+                        if (var1[j].toLowerCase().contains("offset=")) {
+                            filter.setByteOffset(Long.parseLong(var1[j].toLowerCase().replace("offset=", "")));
+                        }
+
+                        if (var1[j].toLowerCase().contains("maxsize=")) {
+                            String value = var1[j].replace("maxSize=", "");
+                            filter.setMaxSize(value.trim());
+                        }
+                    }
+
+                    addFilters(filter);
+                }
+            } catch (RuntimeException e) {
+                LOG.warn("Error configuring filter {}", config);
+            }
+        }
+    }
+
+    private static byte[] hexStringToByteArray(String s) {
+        int len = s.length();
+        byte[] data = new byte[len / 2];
+        for (int i = 0; i < len; i += 2) {
+            data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4)
+                + Character.digit(s.charAt(i + 1), 16));
+        }
+        return data;
+    }
+
+    // Exposed for testing
+    public long translateFileSize(String size) {
```

Comment:
For testing we can just make it package private.

Reply:
True, will fix it.